src/main/java/net/sf/jabref/logic/autocompleter/AutoCompleterFactory.java

```diff
@@ -44,17 +44,20 @@ public AutoCompleterFactory(AutoCompletePreferences preferences, JournalAbbrevia
 
         if (InternalBibtexFields.getFieldExtras(fieldName).contains(FieldProperties.PERSON_NAMES)) {
             return new NameFieldAutoCompleter(fieldName, preferences);
-        } else if (FieldName.CROSSREF.equals(fieldName)) {
+        } else if (InternalBibtexFields.getFieldExtras(fieldName).contains(FieldProperties.SINGLE_ENTRY_LINK)) {
             return new BibtexKeyAutoCompleter(preferences);
-        } else if (FieldName.JOURNAL.equals(fieldName) || FieldName.PUBLISHER.equals(fieldName)) {
+        } else if (InternalBibtexFields.getFieldExtras(fieldName).contains(FieldProperties.JOURNAL_NAME)
+                || FieldName.PUBLISHER.equals(fieldName)) {
             return new JournalAutoCompleter(fieldName, preferences, abbreviationLoader);
         } else {
             return new DefaultAutoCompleter(fieldName, preferences);
         }
     }
 
     public AutoCompleter<String> getPersonAutoCompleter() {
-        return new NameFieldAutoCompleter(Arrays.asList(FieldName.AUTHOR, FieldName.EDITOR), true, preferences);
+        return new NameFieldAutoCompleter(InternalBibtexFields.getAllPublicFieldNames().stream().filter(
```

Comment:
I introduced it in #1720, so easiest thing is to merge that first and then rebase this one.